utils/warnings.ml

```diff
@@ -44,6 +44,9 @@ type upstream_compat_warning =
   | Immediate_void_variant
       (* example: [type t = A of void] is immediate, but
          not after erasure, which boxes void, so it can't be erased. *)
+  | Separability_check
+      (* example: [type packed = | Mk of 'a t [@@unboxed]]
+         where ['a t : value mod non_float]. *)
 
```

Comment:
I'm marginally sad that these examples are in the ml rather than the mli, but this is outside the scope of this PR so feel free to do nothing.